packages/contracts-bedrock/src/Safe/LivenessGuard.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard, GuardManager } from "safe-contracts/base/GuardManager.sol";
+import { ModuleManager } from "safe-contracts/base/ModuleManager.sol";
+import { GetSigners } from "src/Safe/GetSigners.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+contract LivenessGuard is ISemver, BaseGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @notice Emitted when a new set of signers is recorded.
+    /// @param signers An arrary of signer addresses.
+    event SignersRecorded(bytes32 indexed txHash, address[] signers);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice The safe account for which this contract will be the guard.
+    Safe public immutable safe;
+
+    /// @notice A mapping of the timestamp at which an owner last participated in signing a
+    ///         an executed transaction.
+    mapping(address => uint256) public lastLive;
+
+    /// @notice An enumerable set of addresses used to store the list of owners before execution,
+    ///         and then to update the lastSigned mapping according to changes in the set observed
+    ///         after execution.
+    EnumerableSet.AddressSet private ownersBefore;
+
+    /// @notice Constructor.
+    /// @param _safe The safe account for which this contract will be the guard.
+    constructor(Safe _safe) {
+        safe = _safe;
+    }
+
+    /// @notice We use this post execution hook to compare the set of owners before and after.
+    ///         If the set of owners has changed then we:
+    ///         1. Add new owners to the lastSigned mapping
+    ///         2. Delete removed owners from the lastSigned mapping
+    function checkAfterExecution(bytes32, bool) external {
```

Comment:
I suppose this method of tracking the owners before and after is more resilient than intercepting known calls that modify the ownership since there could be a nested call that modifies the ownership

Reply:
Yes, it's less error prone and accounts for the fact that the owners list can be modified in various ways: 
1. `swapOwner`, 
2. `addOwnerWithThreshold`
3. `removeOwner`
4. A delegate call to directly manipulate storage.

This pattern was suggested by @mds1

 this pattern was suggested by @mds1, and is much